migration/load/postgresql/tables/records.py

```diff
@@ -0,0 +1,211 @@
+import csv
+import random
+import uuid
+from datetime import datetime
+
+from invenio_records.dictutils import dict_set
+
+from .base import PostgreSQLTableLoad
+
+# keep track of generated PKs, since there's a chance they collide
+GENERATED_PID_PKS = set()  # TODO: better random logic?
+
+def _pid_pk():
+    while True:
+        # we start at 1M to avoid collisions with existing low-numbered PKs
+        val = random.randint(1_000_000, 2_147_483_647 - 1)
+        if val not in GENERATED_PID_PKS:
+            GENERATED_PID_PKS.add(val)
+            return val
+
+
+def _generate_recid(data):
+    return {
+        "pk": _pid_pk(),
+        "obj_type": "rec",
+        "pid_type": "recid",
+        "status": "R",
+    }
+
+def _generate_uuid(data):
+    return str(uuid.uuid4())
+
+
+class RDMRecordTableLoad(PostgreSQLTableLoad):
```

Comment:
Isn;t this the concept of a "stream"?

Reply:
Not really, the stream will be the bigger picture part. This could be a "substream". Once the code is __fully__ done, this will hopefully be more clear. Same applied to comment below.